app/src/main/java/org/shadowice/flocke/andotp/Activities/MainActivity.java

```diff
@@ -93,6 +95,62 @@
     private ListView tagsDrawerListView;
     private TagsAdapter tagsDrawerAdapter;
     private ActionBarDrawerToggle tagsToggle;
+	private SharedPreferences sharedPreferences;
+
+    /**
+     * The app version code (not the version name!) that was used on the last
+     * start of the app.
+     */
+    private static final String LAST_APP_VERSION = "1";
+
+    /**
+     * Caches the result of {@link #checkAppStart(Context context, SharedPreferences sharedPreferences)}. To allow idempotent method
+     * calls.
+     */
+    private static AppStart appStart = null;
+
+    /**
+     * Finds out started for the first time (ever or in the current version).
+     *
+     * @return the type of app start
+     */
+    public AppStart checkAppStart(Context context, SharedPreferences sharedPreferences) {
+        PackageInfo pInfo;
+
+        try {
+            pInfo = context.getPackageManager().getPackageInfo(
+                    context.getPackageName(), PackageManager.COMPONENT_ENABLED_STATE_DEFAULT);
+            int lastVersionCode = sharedPreferences.getInt(
```

Comment:
We generally try to use the Settings pattern

Reply:
Oh! I see!